src/test/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImplTest.java

```diff
@@ -0,0 +1,344 @@
+        assertStaleCheckExtractionHasCorrectResults(StaleCheckFileFixture.FILE_MIXED_LINES);
+    }
+
+    @Test
+    public void testLoadFileWithMultipleValidRows() throws Exception {
+        assertStaleCheckExtractionHasCorrectResults(StaleCheckFileFixture.FILE_MULTIPLE_VALID_LINES);
+    }
+
+    @Test
+    public void testLoadFileWithMultipleInvalidStatusRows() throws Exception {
+        assertStaleCheckExtractionHasCorrectResults(StaleCheckFileFixture.FILE_MULTIPLE_INVALID_STATUS_LINES);
+    }
+
+    private void assertStaleCheckExtractionHasCorrectResults(StaleCheckFileFixture... fixtures) throws Exception {
+        List<StaleCheckFileFixture> expectedResults = Arrays.asList(fixtures);
+        String[] fileNames = Arrays.stream(fixtures)
+                .map(StaleCheckFileFixture::getBaseFileName)
+                .toArray(String[]::new);
+
+        copyInputFilesAndGenerateDoneFiles(fileNames);
+        staleCheckExtractService.processStaleCheckBatchFiles();
+
+        Map<String, StaleCheckExtractServiceImplTest.StaleCheckFileResult> actualResults = staleCheckExtractService.getFileResults();
+        assertFilesHaveExpectedResults(expectedResults, actualResults);
+        assertDoneFilesWereDeleted(fileNames);
+    }
+
+    private void assertFilesHaveExpectedResults(List<StaleCheckFileFixture> expectedResults, Map<String, StaleCheckFileResult> actualResults) {
+        assertEquals("Wrong number of processed files", expectedResults.size(), actualResults.size());
+        for (int i = 0; i < expectedResults.size(); i++) {
+            StaleCheckFileFixture expectedResult = expectedResults.get(i);
+            StaleCheckFileResult actualResult = actualResults.get(expectedResult.baseFileName);
+            assertFileHasExpectedResults(i, expectedResult, actualResult);
+        }
+    }
+
+    private void assertFileHasExpectedResults(int index, StaleCheckFileFixture expectedResult, StaleCheckFileResult actualResult) {
+        assertEquals("Wrong file-processable state at index " + index, expectedResult.processableFile, actualResult.isProcessableFile());
+        if (expectedResult.processableFile) {
+            List<StaleCheckRowFixture> expectedRows = expectedResult.staleCheckRows;
+            List<StaleCheckRowResult> actualRows = actualResult.getRowResults();
+            assertEquals("Wrong number of processed file rows", expectedRows.size(), actualRows.size());
+
+            for (int i = 0; i < expectedRows.size(); i++) {
+                StaleCheckRowFixture expectedStaleCheckRow = expectedRows.get(i);
+                StaleCheckBatchRow actualStaleCheckRow = actualRows.get(i).staleCheckBatchRow;
+
+                verifyParsedRowMatchesExpectedResult(expectedStaleCheckRow, actualStaleCheckRow);
+            }
+        }
+    }
+
+    private void verifyParsedRowMatchesExpectedResult(StaleCheckRowFixture expectedRow, StaleCheckBatchRow actualRow) {
+        if (expectedRow.valid) {

```

Comment:
Is an extra assertion needed in this method (or in the caller method), to check that the batch row succeeded or failed as expected?